---
title: "R Notebook"
output:
  pdf_document: default
  html_document:
    df_print: paged
---

```{r}
# load mae 
new.mae.filename <- "mae_with-rpkm_additional-data_final.rda"
mae.final.filepath <- file.path("outputs", "01_prepare-datasets", new.mae.filename)
mae <- get(load(mae.final.filepath))
```

This is to perform A/B tests on the bulk datasets using counts, RPKM-counts, logcounts.

# Load/source scripts

```{r, message=F,warning=F,error=F}
base.path <- file.path("code", "07_bulk-scale-abtest_dlpfc-cohort1")
# prep dependencies
source(file.path(base.path, "00_helper-functions.R"))
# prep data
source(file.path(base.path, "01-01_prepare-mae.R")) # prep mae data
source(file.path(base.path, "01-02_prep-s-vector.R"))
# run experiments
source(file.path(base.path, "02-01-01_rse-counts_counts-yz_shared-reference-experiments.R"))
source(file.path(base.path, "02-03-01_rse-rpkm_counts-yz_shared-reference-experiments.R"))
source(file.path(base.path, "02-01-02_rse-counts_logcounts-lutearg_shared-reference-experiments.R"))
source(file.path(base.path, "02-03-02_rse-rpkm_logcounts-lutearg_shared-reference-experiments.R"))
# bind results
source(file.path(base.path, "03_prep-experiment-results_rpkm-trail_cohort1.R"))
```

Filter relevant results.

```{r}
s.set.keep <- c("glial=1,neuron=1", "glial=3,neuron=10")
df.k2 <- df.k2[df.k2$s.set.values %in% s.set.keep,]
# show expt table
expt.table <- table(df.k2$s.set.values, df.k2$experiment.type, 
                    df.k2$bulk.scale.type, df.k2$assay.name.lutearg) %>% 
  as.data.frame()
colnames(expt.table) <- c("s.set.values", "reference.type", "bulk.scale.type", "assay.name.lutearg", "num.results")
knitr::kable(expt.table)
```

# Experiment condition data dictionary

Experiment condition table data dictionary

```{r}
data.dict <- list(z.reference.type = "What type of cell reference atlas was used. 'shared' is the same summary across donors and experiments, 'within' means a new reference was calculated based on each donor's own snRNAseq data.",
                  y.expression.scale = "What type of scaling was used in the bulk RNAseq expression data. 'counts' means unadjusted counts from SPEQeasy, 'RPKM' means the same as 'counts' but with RPKM-normalization, a type of gene/transcript-length normalization.",
                  s.pred.set.name = "Label of the s cell size scale factor set used (see plots).",
                  s.pred.set.values = "Values of the s cell size scale factor set used.",
                  cell.label.type = "Identifier of the cell type labels for experiments.")
data.dict <- do.call(rbind, lapply(seq(length(data.dict)), function(ii){c(names(data.dict)[ii], data.dict[[ii]])}))
knitr::kable(data.dict)
```

# Condition 1 Summaries -- Counts and RPKM-counts

Proportions scatterplots

```{r}
ggplot(df.k2, aes(x = true.neuron, y = neuron)) + theme_bw() +
  geom_point(alpha = 0.5) + geom_abline(slope = 1, intercept = 0) + 
  facet_wrap(~s.set.label*assay.name.lutearg*bulk.scale.type)
```

```{r}
ggplot(df.k2, aes(x = true.neuron, y = neuron, color = bulk.scale.type)) + theme_bw() +
  geom_point(alpha = 0.5) + geom_abline(slope = 1, intercept = 0) + 
  facet_wrap(~s.set.label*assay.name.lutearg)
```
