---
title: "marker_expression_distributions"
author: "Sean Maden"
date: "2023-10-30"
bibliography: bibliography.bib
output:
  html_document: default
  pdf_document: default
---

```{r setup, include=FALSE}
libraryVector <- c("ggplot2", "reshape2", "dplyr", "ComplexHeatmap", "UpSetR")
sapply(libraryVector, library, character.only = TRUE)
knitr::opts_chunk$set(echo = TRUE)
setwd("..")
setwd("..")
load("./env/03_k5/01_run_script.RData")
```

# Purpose

This notebook assesses the @monaco_rna-seq_2019 gene expression scales and 
performs marker summaries for cell type refactoring.

## Scale importance

For gene expression experiments, the bulk expression scaling affects 
deconvolution performance.

The @monaco_rna-seq_2019 study published gene expression in transcripts per 
million (TPM) scale. Analyses were performed using log2(TPM+1) scale.

## Published markers

Markers were published for immune and blood cell types and were studied in 
peripheral blood mononuclear cells (PBMCs).

The initial cell types included over 60 cell types profiled by flow cytometry.
The study then identified 17 cell types with markers in either RNA-seq or 
microarray, with 1,294 markers in the former.

# Marker summaries

Marker expression by cell type

## Quantiles filter

Quantile filter of greater than or equals to 90% was defined, or the top 130 
markers for each cell type were taken.

## Show counts

Marker abundance after quantiles filter.

```{r}

ggplot(
  dfPlotHeatmapLog2TpmRef, 
  aes(x = cellType, y = markerCount, fill = isTypeMarker)) + 
  geom_bar(stat = "identity") +
  theme(axis.text.x = element_text(angle = 45, hjust = 1)) +
  geom_text(aes(label = markerCount), vjust=1)

upset()

```

## Annotated heatmap

Show mappings table

```{r}
mappingsTable
```

Show heatmap with markers labeled

```{r}
Heatmap(
  scaleLog2TpmReference, 
  name = "Z",
  right_annotation = rowAnnotation(
    cellType1 = markerColors$cellType2,
    cellType2 = markerColors$cellType1
  ),
  show_row_dend = FALSE
)

```


# Getting markers

## Original expression

```{r}
Heatmap(tpmReference, name = "TPM")
Heatmap(scaleTpmReference, name = "Z TPM")
Heatmap(log2TpmReference, name = "log2 TPM")
Heatmap(scaleLog2TpmReference, name = "Z log2 TPM")
```

## Quantile expression

Shows the threshold heatmap.

```{r}



```

## Quantile genes by type

```{r}

ggplot()

```

# Get the binary plot for TPM reference

## TPM reference

```{r}



```

Plot

```{r}
```

Save

```{r}
```

# Cluster

```{r}



```

```{r}

#dfplot <- melt(clusterTpmReference$x)
#ggplot(clusterTpmReference$x, aes(x = PC1, y = PC2, color = )) + geom_point()

```

# Summary statistics

## TPM

```{r}
summary(tpmZref)
```

## Log2 TPM

```{r}
summary(log2TpmZref)
```

## Z score (from TPM)

```{r}
summary(zscoreTpm)
```

## Z score (from log2 TPM)

```{r}
summary(zscoreLog2TpmZref)
```

# Density plots

```{r}
```

## TPM

```{r}
densityPlot(zref)
```

## Log2 TPM

```{r}
densityPlot(log2TpmZref)
```

## Z score (from TPM)

```{r}
densityPlot(zscoreTpm)
```

## Z score (from log2 TPM)

```{r}
densityPlot(zscoreLog2TpmZref)
```

# Session info

```{r}
sessionInfo()
```

# Works cited
