---
title: "marker_expression_distributions"
author: "Sean Maden"
date: "2023-10-30"
bibliography: bibliography.bib
output:
  html_document: default
  pdf_document: default
---

```{r setup, include=FALSE}
libraryVector <- c("ggplot2", "reshape2", "dplyr", "ComplexHeatmap", "UpSetR")
sapply(libraryVector, library, character.only = TRUE)
knitr::opts_chunk$set(echo = TRUE)
setwd("..")
setwd("..")
load("./env/03_k5/01_run_script.RData")
```

# Purpose

This notebook assesses the @monaco_rna-seq_2019 gene expression scales and 
performs marker summaries for cell type refactoring.

## Scale importance

For gene expression experiments, the bulk expression scaling affects 
deconvolution performance.

The @monaco_rna-seq_2019 study published gene expression in transcripts per 
million (TPM) scale. Analyses were performed using log2(TPM+1) scale.

## Published markers

Markers were published for immune and blood cell types and were studied in 
peripheral blood mononuclear cells (PBMCs).

The initial cell types included over 60 cell types profiled by flow cytometry.
The study then identified 17 cell types with markers in either RNA-seq or 
microarray, with 1,294 markers in the former.

# Markers by cell type

## Show counts

```{r}
listQuantiles <- getQuantileTablesFromReferenceExpression(
  log2TpmReference, "log2 TPM", 10, seq(0,1,0.1)
)
dfTall <- melt(listQuantiles$booleanTable)
dfPlot <- dfTall %>% 
  group_by(Var2) %>% 
  count(value) %>% 
  mutate(prop = prop.table(n))
colnames(dfPlot) <- 
  c("cellType", "isTypeMarker", "markerCount", "markerProportion")
ggplot(dfPlot, aes(x = cellType, y = markerCount, fill = isTypeMarker)) + 
  geom_bar(stat = "identity") +
  theme(axis.text.x = element_text(angle = 45, hjust = 1)) +
  geom_text(aes(label = markerCount), vjust=1)
```

## Annotated heatmap

Show mappings table

```{r}
mappingsTable
```

Show heatmap with markers labeled

```{r}
# map plot cols to cell types
# use 2 cell type mappings
#
mappingsTable$colorLabel1 <- mappingsTable$celltype1 %>%
  as.factor() %>% as.numeric()
mappingsTable$colorLabel2 <- mappingsTable$celltype2 %>%
  as.factor() %>% as.numeric()

# get markers vector with colors
markerColors <- data.frame(
  marker=rownames(scaleLog2TpmReference)
)
markerColors$cellType <- 
  markerColors$markerColor1 <- 
  markerColors$markerColor2 <- "NA"
for(type in colnames(scaleLog2TpmReference)){
  typeMarkerNames <- dfTall[dfTall$Var2==type & dfTall$value==TRUE,]$Var1
  markerColors[markerColors$marker %in% typeMarkerNames,]$cellType1 <- type
  markerColors[markerColors$marker %in% typeMarkerNames,]$cellType1 <- 
    mappingsTable[mappingsTable$celltype1==type,]$celltype2
  markerColors[markerColors$marker %in% typeMarkerNames,]$markerColor1 <- 
    mappingsTable[mappingsTable$celltype1==type,]$colorLabel1
  markerColors[markerColors$marker %in% typeMarkerNames,]$markerColor2 <- 
    mappingsTable[mappingsTable$celltype1==type,]$colorLabel2
}

Heatmap(
  scaleLog2TpmReference, 
  name = "Z",
  right_annotation = rowAnnotation(
    cellType1 = markerColors$markerColor1,
    cellType2 = markerColors$markerColor2
  )
)

Heatmap(
  scaleLog2TpmReference, 
  name = "Z",
  right_annotation = rowAnnotation(
    cellType1 = markerColors$cellType2,
    cellType2 = markerColors$cellType1
  )
)

```


# Getting markers

## Original expression

```{r}
Heatmap(tpmReference, name = "TPM")
Heatmap(scaleTpmReference, name = "Z TPM")
Heatmap(log2TpmReference, name = "log2 TPM")
Heatmap(scaleLog2TpmReference, name = "Z log2 TPM")
```

## Quantile expression

Shows the threshold heatmap.

```{r}

getQuantileTablesFromReferenceExpression(
  tpmReference, "TPM", 10, seq(0,1,0.1)
)[[3]]

getQuantileTablesFromReferenceExpression(
  scaleTpmReference, "Z TPM", 10, seq(0,1,0.1)
)[[3]]

getQuantileTablesFromReferenceExpression(
  log2TpmReference, "log2 TPM", 10, seq(0,1,0.1)
)[[3]]

getQuantileTablesFromReferenceExpression(
  scaleLog2TpmZref, "Z log2 TPM", 10, seq(0,1,0.1)
)[[3]]

```

## Quantile genes by type

```{r}

ggplot()

```

# Get the binary plot for TPM reference

## TPM reference

```{r}

listQuantileTables <- 
  getQuantileTablesFromReferenceExpression(tpmReference)

# test
identical(
  ifelse(
    listQuantileTables$booleanTable[1,seq(10)], 1, 0),
  listQuantileTables$numericTable[1,seq(10)]
) # TRUE

```

Plot

```{r}


# plot the numeric table

heatmap(
  as.numeric(referenceQuantile))

```

Save

```{r}
```

# Cluster

```{r}

clusterTpmReference <- prcomp(tpmReference)
clusterTpmLog2Reference <- prcomp(log2TpmReference)
clusterScaleTpmReference <- prcomp(scaleTpmReference)
clusterScaleTpmLog2Reference <- prcomp(scaleLog2TpmZref)

```

```{r}

dfplot <- melt(clusterTpmReference$x)
dfplot$cellType <- 

ggplot(clusterTpmReference$x, aes(x = PC1, y = PC2, color = )) + geom_point()

```

# Summary statistics

## TPM

```{r}
summary(tpmZref)
```

## Log2 TPM

```{r}
summary(log2TpmZref)
```

## Z score (from TPM)

```{r}
summary(zscoreTpm)
```

## Z score (from log2 TPM)

```{r}
summary(zscoreLog2TpmZref)
```

# Density plots

```{r}
meanDifferences <- function(dfplot){
  cellTypes <- colnames(dfplot)
  
  
  
  markersVector <- rownames(dfplot)
  dfMeanAllTypes <- apply(dfplot, 2, mean)
  
  
  
  lapply(markersVector, function(marker){
    
    mean(
      
    )
    
    dfPlotOut <- dfplot[,!colnames(dfplot)==cellType]
    meanOut <- apply(dfPlotDifference,2,mean)
    meanIn <- mean(dfplot[,cellType])
    
    
    
  })
  
  
}

densityPlot <- function(dfplot){
  dfplot <- melt(dfplot)
  dfplot[,2] <- as.numeric(dfplot[,2])
  
  # get pairwise mean differences
  
  plotAllMarkers <- ggplot(dfplot, aes(x = variable, y = value)) + 
    geom_density() +
    theme(axis.text.x = element_text(angle = 45, hjust = 1))
  
  
  
  return(listPlots)
}
```

## TPM

```{r}
densityPlot(zref)
```

## Log2 TPM

```{r}
densityPlot(log2TpmZref)
```

## Z score (from TPM)

```{r}
densityPlot(zscoreTpm)
```

## Z score (from log2 TPM)

```{r}
densityPlot(zscoreLog2TpmZref)
```

# Session info

```{r}
sessionInfo()
```

# Works cited
