---
title: "Plot results of bulk A/B tests with and without cell size scaling"
output:
  pdf_document: default
  html_document:
    df_print: paged
---

This is to perform A/B tests on the bulk datasets using counts, RPKM-counts, logcounts.

```{r}
knitr::opts_chunk$set(fig.width=10, fig.height=10)
libv <- c("ggplot2")
sapply(libv, library, character.only = T)
```
```{r}
setwd("..")
setwd("..")
load("./env/05_bulk/01_run_script.RData")
```

# Experiment summaries

Overall experiment design table with conditions tested.

```{r}
expt <- data.frame(z.reference.type = c("shared", "within", "shared", "within"),
                   y.expression.scale = c("counts", "counts", "RPKM", "RPKM"))
expt <- do.call(rbind, lapply(seq(length(list.s.pred)), function(ii){
  expt.iter <- expt
  expt.iter$s.pred.set.name <- names(list.s.pred)[ii]
  expt.iter$s.pred.set.values <- paste0(names(list.s.pred[[ii]]), ":", 
                                        list.s.pred[[ii]],collapse = ";")
  expt.iter
}))
expt$cell.label.type <- "k2"

# table visualization
knitr::kable(expt)
```

Experiment condition table data dictionary

```{r}
data.dict <- list(z.reference.type = "What type of cell reference atlas was used. 'shared' is the same summary across donors and experiments, 'within' means a new reference was calculated based on each donor's own snRNAseq data.",
                  y.expression.scale = "What type of scaling was used in the bulk RNAseq expression data. 'counts' means unadjusted counts from SPEQeasy, 'RPKM' means the same as 'counts' but with RPKM-normalization, a type of gene/transcript-length normalization.",
                  s.pred.set.name = "Label of the s cell size scale factor set used (see plots).",
                  s.pred.set.values = "Values of the s cell size scale factor set used.",
                  cell.label.type = "Identifier of the cell type labels for experiments.")
data.dict <- do.call(rbind, lapply(seq(length(data.dict)), function(ii){c(names(data.dict)[ii], data.dict[[ii]])}))
knitr::kable(data.dict)
```

# Experiment condition summaries

Proportions scatterplots.

```{r}
new.plot.scatter <- ggplot(df.k2, 
                           aes(x = true.neuron, y = neuron, color = bulk.scale.type)) + 
  theme_bw() + geom_point(alpha = 0.5) + geom_abline(slope = 1, intercept = 0) + 
  facet_wrap(~experiment.type*assay.name.lutearg*s.set.label)

new.plot.scatter
```

Save plot

```{r}
setwd("..")
setwd("..")
new.plot.path <- "./figures/05_bulk/figs2_scatter.jpg"
jpeg(new.plot.path, width = 6, height = 5, res = 400, units = "in")
new.plot.scatter
dev.off()
```

# Jittered boxplots

Absolute errors, organized by S cell size scale factor set ID.

```{r}
new.plot.jitterbox <- ggplot(df.k2, aes(x = experiment.type, y = abs.error.neuron)) + 
  geom_jitter(alpha = 0.5) + geom_boxplot(alpha = 0, color = "cyan") +
  facet_wrap(~assay.name.lutearg) + theme_bw() +
  theme(axis.text.x = element_text(angle = 45, hjust = 1))

new.plot.jitterbox
```

Save plot

```{r}
setwd("..")
setwd("..")
new.plot.path <- "./figures/05_bulk/figs2b_jitterbox1.jpg"
jpeg(new.plot.path, width = 6, height = 5, res = 400, units = "in")
new.plot.jitterbox
dev.off()
```

# Errors by sample id and condition

Absolute errors, organized by sample ID.

```{r}
new.plot.jitter.cond <- ggplot(df.k2, aes(x = bulk.sample.condition, y = abs.error.neuron, color = sample.id)) + 
  geom_jitter(alpha = 0.3) + geom_boxplot(alpha = 0, color = "black") + theme_bw() +
  facet_wrap(~experiment.type) + theme(axis.text.x = element_text(angle = 45, hjust = 1))

new.plot.jitter.cond
```

Save plot

```{r}
setwd("..")
setwd("..")
new.plot.path <- "./figures/05_bulk/figs2c_jitterbox2_condition.jpg"
jpeg(new.plot.path, width = 6, height = 4, res = 400, units = "in")
new.plot.jitter.cond
dev.off()
```

