---
title: "04_simulations"
author: "Sean Maden"
date: "2023-11-22"
output:
  pdf_document: default
  html_document: default
---

Results from pseudobulk simulations using Plasmablast and Non-plasmablast cell 
proportions from Monaco et al 2019.

```{r setup, include=FALSE}
libv <- c("lute", "ggplot2", "gridExtra", "reshape2", "ggforce")
sapply(libv, library, character.only = TRUE)
setwd("..")
setwd("..")
load("./env/01_pseudobulk/01_read_script.RData")
load("./env/01_pseudobulk/04_simulation.RData")
knitr::opts_chunk$set(echo = FALSE)
```

# Background

## Simulations 

This simulation used randomized single-cell expression. Two cell type labels 
were used, either "Plasmablasts" for the larger cell type category, or 
"Non-plasmablasts" for the smaller cell type category.

In addition, the following data were drawn from the experiment: 1. Proportions 
were calculated for each category from flow cytometry proportions; 2. Mean 
expression was calculated from bulk RNA-seq TPM data at ABIS-seq marker genes.

## Data summary

# Pseudobulk preparation

Simulations were run for 12 (N = `r length(unique(dfPlotTall$sample.id))`) 
donors with available known cell type proportions from flow cytometry. In total, 
data from 17 (N = `r ncol(dfPlotTall[,!colnames(dfPlotTall) %in% c("sample.id", "type")])`) cell types were used, where 16 were aggregated under 
"non-plasmablast", and "plasmablast" was retained.

# Pseudobulk properties

Pseudobulking used 2 (N = `r ncol(dfPseudobulk[,grepl("plasmablast", colnames(dfPseudobulk))])`) cell types and 12 (N = `r length(unique(dfPseudobulk$sample.id))`) samples.

## Cell size calculations

Cell size scale factors obtained as the library sizes (column sums) from the 
RNA-seq reference in Monaco et al 2019 (Table S5). Binarized cell sizes 
(Plasmablasts, Non-plasmablasts) for Non-plasmablasts were taken as the median 
of cell sizes.

# Scatter plots of simulation outcomes

Reporting pseudobulk results, at 3 zoom levels.

```{r}
dfPlot <- dfPseudobulkA
plot1 <- ggplot(dfPseudobulkA, aes(x = true.plasmablasts, y = Plasmablasts)) + 
  geom_point(size = 4, alpha = 0.5) + geom_abline(slope = 1, intercept = 0) +
  facet_wrap(~condition) + xlim(0, 0.15) + ylim(0, 0.15) +
  ylab("Predicted") + xlab("True")
plot2 <- ggplot(dfPseudobulkA, aes(x = true.plasmablasts, y = Plasmablasts)) + 
  geom_point(size = 4, alpha = 0.5) + geom_abline(slope = 1, intercept = 0) +
  facet_wrap(~condition) + xlim(0, 0.5) + ylim(0, 0.5) +
  ylab("Predicted") + xlab("True")
plot3 <- ggplot(dfPseudobulkA, aes(x = true.plasmablasts, y = Plasmablasts)) + 
  geom_point(size = 4, alpha = 0.5) + geom_abline(slope = 1, intercept = 0) +
  facet_wrap(~condition) + xlim(0, 1) + ylim(0, 1) +
  ylab("Predicted") + xlab("True")
plot1
plot2
plot3
```

```{r, fig.height = 15, fig.width = 9}
# multiplot
textSize <- 30
plot1 <- 
  plot1 + xlab("") + ylab("") + ggtitle("Zoom1") + 
  theme(text = element_text(size=textSize),
        axis.text.x = element_text(angle=45,hjust=1))
plot2 <- 
  plot2 + xlab("") + ylab("Predicted") + ggtitle("Zoom2") + 
  theme(text = element_text(size=textSize),
        axis.text.x = element_text(angle=45,hjust=1))
plot3 <- 
  plot3 + xlab("") + ylab("True") + ggtitle("Zoom3") + 
  theme(text = element_text(size=textSize),
        axis.text.x = element_text(angle=45,hjust=1))
grid.arrange(plot1, plot2, plot3, ncol = 1, top = "Plasmablasts")
```

# Save

```{r}
setwd("..")
setwd("..")
list.files("figures")
```

```{r}

setwd("..")
setwd("..")

dfPlot <- dfPseudobulkA
dfPlot$condition2 <- 
  ifelse(dfPlot$condition=="scaled", "withscale", "noscale")
dfPlot$condition2 <- 
  factor(dfPlot$condition2, levels = c("noscale", "withscale"))

# zoom level 1, axis maximum 0.5

jpeg("figures/01_pseudobulk/supplemental_figure2a.jpg", 
     width = 4, height = 2.5, units = "in", res = 400)

ggplot(dfPlot, aes(x = true.plasmablasts, y = Plasmablasts)) + 
  geom_point(size = 4, alpha = 0.5) + theme_bw() +
  geom_abline(slope = 1, intercept = 0) +
  facet_wrap(~condition2) + xlim(0, 0.15) + ylim(0, 0.15) +
  ylab("Predicted") + xlab("Known") +
  theme(axis.text.x = element_text(angle = 45, hjust = 1))

dev.off()

# zoom level 2, axis maximum 1

jpeg("figures/01_pseudobulk/supplemental_figure2b.jpg", 
     width = 4, height = 2.5, units = "in", res = 400)

ggplot(dfPlot, aes(x = true.plasmablasts, y = Plasmablasts)) + 
  geom_point(size = 4, alpha = 0.5) + theme_bw() +
  geom_abline(slope = 1, intercept = 0) +
  facet_wrap(~condition2) + xlim(0, 1) + ylim(0, 1) +
  ylab("Predicted") + xlab("Known") +
  theme(axis.text.x = element_text(angle = 45, hjust = 1))

dev.off()

```

# Scatterplots with 2 cell types from K2 results

Make plot from pseudobulk results.

```{r}
dfi <- dfPseudobulkA
dfi1 <- data.frame(
  true = dfi$true.plasmablasts,
  pred = dfi$Plasmablasts,
  sample.id = dfi$sample.id,
  type = dfi$condition,
  celltype = "plasmablast"
)
dfi2 <- data.frame(
  true = dfi$true.not.plasmablasts,
  pred = dfi$'Non-plasmablast',
  sample.id = dfi$sample.id,
  type = dfi$condition,
  celltype = "non-plasmablast"
)
dfp <- rbind(dfi1, dfi2) |> as.data.frame()
```

Preview plots.

```{r}
library(ggforce)
cellTypeColor1 <- "blue"
cellTypeColor2 <- "red"
idLineColor <- "orange"

# get plot objects
new.plot.k2.noline <- ggplot(dfp, 
                   aes(x = true, y = pred, colour = celltype)) + 
  geom_point(size = 4, alpha = 0.3) + 
  geom_abline(slope = 1, intercept = 0) + 
  geom_hline(yintercept = 0.5) + 
  geom_vline(xintercept = 0.5) + theme_bw() +
  xlab("Known") + ylab("Predicted") +
  xlim(0, 1) + ylim(0, 1) + facet_wrap(~type) +
  theme(axis.text.x = element_text(angle = 45, hjust = 1)) +
  scale_color_manual(values = c(cellTypeColor2, cellTypeColor1))
new.plot.k2.withline <-
  new.plot.k2.noline + geom_line(aes(group=sample.id), color = idLineColor)

new.plot.k2.noline
new.plot.k2.withline
```

Try more complicated visualizations with facet zooms.

```{r}
new.plot.k2.noline + facet_zoom(ylim = c(0, 0.2), xlim = c(0, 0.2),
                                split = TRUE)
new.plot.k2.withline + facet_zoom(ylim = c(0, 0.2), xlim = c(0, 0.2),
                                  split = TRUE)
```

```{r}
# example without line
plot.zoom.1 <- ggplot(dfp[dfp$type=="scaled",], 
                   aes(x = true, y = pred, colour = celltype)) + 
  geom_point(size = 4, alpha = 0.3) + 
  geom_abline(slope = 1, intercept = 0) + 
  geom_hline(yintercept = 0.5) + 
  geom_vline(xintercept = 0.5) + theme_bw() +
  xlab("Known") + ylab("Predicted") +
  xlim(0, 1) + ylim(0, 1) + 
  theme(axis.text.x = element_text(angle = 45, hjust = 1)) +
  scale_color_manual(values = c(cellTypeColor2, cellTypeColor1)) +
  facet_zoom(ylim = c(0, 0.1), xlim = c(0, 0.1), split = FALSE,
             show.area = FALSE) +
  ggtitle("withscale")

plot.zoom.2 <- ggplot(dfp[dfp$type=="unscaled",], 
                   aes(x = true, y = pred, colour = celltype)) + 
  geom_point(size = 4, alpha = 0.3) + 
  geom_abline(slope = 1, intercept = 0) + 
  geom_hline(yintercept = 0.5) + 
  geom_vline(xintercept = 0.5) + theme_bw() +
  xlab("Known") + ylab("Predicted") +
  xlim(0, 1) + ylim(0, 1) +
  theme(axis.text.x = element_text(angle = 45, hjust = 1)) +
  scale_color_manual(values = c(cellTypeColor2, cellTypeColor1)) +
  facet_zoom(ylim = c(0, 0.1), xlim = c(0, 0.1), split = FALSE,
             show.area = FALSE) +
  ggtitle("noscale")

grid.arrange(plot.zoom.1, plot.zoom.2, ncol = 2)

# example with line
plot.zoom.1.withline <- plot.zoom.1 + geom_line(aes(group = sample.id), 
                                                col=idLineColor)
plot.zoom.2.withline <- plot.zoom.2 + geom_line(aes(group = sample.id), 
                                                col=idLineColor)

grid.arrange(plot.zoom.2, plot.zoom.1, ncol = 2)
```

Save to JPEG.

```{r}
setwd("..")
setwd("..")
jpeg("./figures/01_pseudobulk/fig2_pbmc-k2-result-withline.jpg", 
     width = 6.5, height = 3, units = "in", res = 400)
new.plot.k2.withline
dev.off()

jpeg("./figures/01_pseudobulk/fig2_pbmc-k2-result-noline.jpg", 
     width = 6.5, height = 3, units = "in", res = 400)
new.plot.k2.noline
dev.off()

lm <- matrix(c(rep(1,6),rep(2,8)),nrow=1)
jpeg("./figures/01_pseudobulk/fig2_pbmc-k2-result-withline_zooms.jpg", 
     width = 10, height = 3, units = "in", res = 400)
grid.arrange(plot.zoom.2.withline +
               theme(
                 legend.position = "none",
                 axis.title.x = element_blank(),
                 axis.title.y = element_blank()
               ),
             plot.zoom.1.withline + 
               theme(
                 axis.title.x = element_blank(),
                 axis.title.y = element_blank()
              ),
             layout_matrix = lm, ncol = 2,
             bottom = "Known", left = "Predicted")
dev.off()
```

# Session info

```{r}
sessionInfo()
```